mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/BlockMatrix.scala

```diff
@@ -172,6 +172,64 @@ class BlockMatrix(
     assert(cols <= nCols, s"The number of columns $cols is more than claimed $nCols.")
   }
 
+  def validate: Unit = {
+    logDebug("Validating BlockMatrix...")
+    // check if the matrix is larger than the claimed dimensions
+    try {
+      estimateDim()
+      logDebug("BlockMatrix dimensions are okay...")
+    } catch {
```

Comment:
Not necessary to catch exceptions. `estimateDim` should throw exception with clean message.